restapi/src/main/java/io/stargate/web/docsapi/service/ReactiveDocumentService.java

```diff
@@ -562,24 +585,36 @@ public ReactiveDocumentService(
    */
   public Maybe<DocumentResponseWrapper<? extends JsonNode>> executeBuiltInFunction(
       DocumentDB db,
-      String keyspace,
+      String namespace,
       String collection,
       String id,
       ExecuteBuiltInFunction funcPayload,
-      List<PathSegment> path,
+      List<String> path,
       ExecutionContext context) {
     return Maybe.defer(
         () -> {
-          List<String> pathString =
-              path.stream().map(seg -> seg.getPath()).collect(Collectors.toList());
-          Maybe<JsonNode> result = null;
+          // process path
+          List<String> processedPath = processSubDocumentPath(path);
+
+          // note that currently all built in-functions require auth for modify and delete
+          // please adapt if needed in future
           BuiltInApiFunction function = BuiltInApiFunction.fromName(funcPayload.getOperation());
+          authorizeWrite(db, namespace, collection, Scope.MODIFY, Scope.DELETE);
+
+          // based on type switch the result
+          Maybe<JsonNode> result = null;
           if (function == BuiltInApiFunction.ARRAY_PUSH) {
-            result =
-                handlePush(
-                    db, keyspace, collection, id, funcPayload.getValue(), pathString, context);
+            // explicit null check to ensure we have a value to push
+            if (null == funcPayload.getValue()) {
+              throw new ErrorCodeRuntimeException(ErrorCode.DOCS_API_INVALID_JSON_VALUE);
+            }
+
+            // TODO Eric what's the desired object here and how would this me mapped anyway?
+            //  should we fix the value in the Function payload to JsonNode?
```

Comment:
@tatu-at-datastax @EricBorczuk Setting the `JsonNode value` in the [ExecuteBuiltInFunction](https://github.com/stargate/stargate/blob/576cc162a0a3592b0a59720666e76108888b7ae0/restapi/src/main/java/io/stargate/web/docsapi/models/dto/ExecuteBuiltInFunction.java) cannot distinguish between null and missing node.. There is an issue here that explains it [FasterXML/jackson-databind#3214](https://github.com/FasterXML/jackson-databind/issues/3214)

So I would like to have it, but seems that it's not working at least at the moment.. If you have any other idea I would be up for a fix.. Until then, we can keep it like this I guess.. We want to allow pushing of `null`s, but we also want that to be explicit and report bad request in case value is not set..

Reply:
Yeah I guess for now it has to remain this way, then